consoles/VNC.pm

```diff
@@ -921,6 +936,65 @@ sub _receive_zrle_encoding ($self, $x, $y, $w, $h) {
     return $res;
 }
 
+# wrapper to make testing easier
+sub _read_socket ($socket, $data, $data_len, $offset) {
+    return read($socket, $data, $data_len, $offset);
+}
+
+sub _receive_tight_encoding ($self, $x, $y, $w, $h) {
```

Comment:
As this file consoles/VNC.pm is getting quite big I would prefer if we can split the code. Do you think this new method or some other code could fit elsewhere and then include here?

Reply:
I guess `_receive_*_encoding` could be placed in another (new) file. But that won't be pretty, especially since `_receive_zrle_encoding` and `_receive_ikvm_encoding` use few more fields from the VNC object, so at least those will need to get whole `$self` to keep it sane.